bop_toolkit_lib/inout.py

```diff
@@ -139,22 +140,56 @@ def load_cam_params(path):
 
 
 def _camera_as_numpy(camera):
+    """Convert fields from scene camera from native python to numpy.
+
+    See docs/bop_datasets_format.md for details.
+    Note: "cam_K" and "cam_model" are mutually exclusive and raise a ValueError.
+
+    :param camera: Dictionnary containing
+    - 'cam_R_w2c': orientation of world frame in camera frame
+    - 'cam_t_w2c': position of world frame in camera frame
+    - 'cam_K': IF BOP19 format, camera pinhole intrinsics parameters
+    - 'cam_model': IF BOP24 format
+    :return: Dictionnary with same keys and flat lists field as numpy arrays 
+    """
+    if "cam_K" in camera and "cam_model" in camera:
+        raise ValueError("Only one of 'cam_K', 'cam_model' field should be present in a scene camera configuration")
     if "cam_K" in camera.keys():
         camera["cam_K"] = np.array(camera["cam_K"], np.float64).reshape((3, 3))
     if "cam_R_w2c" in camera.keys():
         camera["cam_R_w2c"] = np.array(camera["cam_R_w2c"], np.float64).reshape((3, 3))
     if "cam_t_w2c" in camera.keys():
         camera["cam_t_w2c"] = np.array(camera["cam_t_w2c"], np.float64).reshape((3, 1))
+    if "cam_model" in camera:
+        camera["cam_model"]["projection_params"] = np.array(camera["cam_model"]["projection_params"], np.float64)
+        # TODO: convert extrinsics too
```

Comment:
is `convert extrinsics too` required for the evaluation?

Reply:
Will remove the comment as not relevant at the moment